src/alloc_bytes.rs

```diff
@@ -0,0 +1,113 @@
+use std::alloc;
+use std::alloc::Layout;
+use std::slice;
+use std::borrow::Cow;
+
+use rustc_target::abi::{Align, Size};
+use rustc_middle::mir::interpret::AllocBytes;
+
+/// Allocation bytes that explicitly handle the layout of the data they're storing.
+/// This is necessary to interface with native code that accesses the program store in Miri.
+#[derive(Debug)]
+#[derive(PartialEq, Eq, Hash)] // TODO Necessary?
```

Comment:
They should not be necessary. What happens if you remove them?

IIRC we said that https://github.com/rust-lang/rust/pull/124492 should change [this line](https://github.com/rust-lang/rust/blob/7be47b219df42cba96560a97de707967cca8c762/compiler/rustc_middle/src/mir/interpret/allocation.rs#L32) to remove the `Eq + PartialEq + Hash`.

Reply:
Currently when I remove them it complains about it at `impl AllocBytes for MiriAllocBytes`. I'll remove this line and incorporate the change of `AllocBytes` inside Rust itself in the other PR.

Reply:
Yeah, this will have to wait until https://github.com/rust-lang/rust/pull/124492 lands. So let's finish that PR first. :)